src/modules/auth/credentials/BulkAssignmentContainer.js

```diff
@@ -0,0 +1,460 @@
+			setTimeout(() => setShowTenantsContentLoader(true), 500);
+		}
+		if (limit > 0) {
+			retrieveTenants();
+		}
+	}, [tenantsPage, tenantsFilter, tenantsLimit]);
+
+	/* Actual modified data (this data set was compared compared with `selectedCredentials` and `assigned: true` was added to matching credentials. Assigned:true property
+	disables the "+"" button) to be displayed in Credentials list data table */
+	const datatableCredentialsData = useMemo(() => {
+		let credentialsTableData = [];
+		if (data) {
+			data.map((credObj) => {
+				/* we are maping over `data` (state used for Credentials). Each element of `data` state is an object with property `_id` (among others).
+				We are trying to match the value of this property with any selectedCredentials state's element's `_id` property.
+				If a match is found, a key value pair of `assigned: true` is assigned to that object */
+				let matchedObj = selectedCredentials.find(obj => obj._id === credObj._id);
+				if (matchedObj) {
+					matchedObj['assigned'] = true;
+					credentialsTableData.push(matchedObj);
+				} else {
+					credObj['assigned'] = false;
+					credentialsTableData.push(credObj);
+				}
+			})
+		}
+		return credentialsTableData
+	}, [data, selectedCredentials]);
+
+	const datatableTenantsData = useMemo(() => {
+		let tenantsTableData = [];
+		if (tenants?.length > 0) {
+			tenants.map((tenantObj) => {
+				/* we are maping over `tenants` (state used for Tenants). Each element of `tenants` state is an object with property `_id` (among others).
+				We are trying to match the value of this property with any selectedTenants state's element's `_id` property.
+				If a match is found, a key value pair of `assigned: true` is assigned to that object */
+				let matchObj = selectedTenants.find(obj => obj._id === tenantObj._id);
+				if (matchObj) {
+					matchObj['assigned'] = true;
+					tenantsTableData.push(matchObj);
+				} else {
+					tenantObj['assigned'] = false;
+					tenantsTableData.push(tenantObj);
+				}
+			})
+		}
+		return tenantsTableData
+	}, [tenants, selectedTenants]);
+
+	// fetches Credentials Data from server
+	const retrieveData = async () => {
+		setLoading(true);
+		try {
+			let response = await SeaCatAuthAPI.get("/credentials", {params: {p:page, i: limit, f: credentialsFilter}});
+			if (response.data.result !== "OK") {
+				throw new Error(t("BulkAssignmentContainer|Failed to fetch credentials"));
+			};
+			setData(response.data.data);
+			setCount(response.data.count);
+		} catch(e) {
+			console.error(e);
+			// error message for unauthorized access
+			if (e.response.status === 401) {
+				props.app.addAlert("warning", t("BulkAssignmentContainer|Can't fetch credentials, you don't have rights to display it"), 30);
+				return;
+			};
+			props.app.addAlert("warning", `${t("BulkAssignmentContainer|Failed to fetch credentials")}. ${e?.response?.data?.message}`, 30);
+			setLoading(false);
```

Comment:
@petrKavulok What if response status will be 401? Then loading will still be `true` . I suggest to move it above the if statement